extensions/resteasy-reactive/rest-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/BasicServerJacksonMessageBodyWriter.java

```diff
@@ -20,23 +22,42 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.ObjectWriter;
 
+import io.quarkus.resteasy.reactive.jackson.runtime.mappers.JacksonMapperUtil;
+
 public class BasicServerJacksonMessageBodyWriter extends ServerMessageBodyWriter.AllWriteableMessageBodyWriter {
 
     private final ObjectWriter defaultWriter;
+    private final Map<Type, ObjectWriter> genericWriters = new ConcurrentHashMap<>();
 
     @Inject
     public BasicServerJacksonMessageBodyWriter(ObjectMapper mapper) {
         this.defaultWriter = createDefaultWriter(mapper);
     }
 
+    private ObjectWriter getWriter(Type genericType, Object value) {
+        // make sure we properly handle polymorphism in generic collections
+        if (genericType != null && value != null) {
+            ObjectWriter writer = genericWriters.get(genericType);
+            if (writer == null) {
+                // no cached writer for that type. Compute it.
+                writer = JacksonMapperUtil.getGenericWriter(genericType, this.defaultWriter);
+                genericWriters.put(genericType, writer);
+            }
+            return writer;
```

Comment:
This operation is not atomic and it may happen that multiple writers could be unnecessarily generated for the same type. I believe that this whole block could be replaced with a one-liner, something like:

`return genericWriters.computeIfAbsent(genericType, type -> JacksonMapperUtil.getGenericWriter(type, defaultWriter));`